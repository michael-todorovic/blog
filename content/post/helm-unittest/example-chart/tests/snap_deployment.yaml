suite: Deployment
templates:
  - deployment.yaml
  - service.yaml
tests:
  - it: test base
    values:
      - ./values/base.yaml
    asserts:
      - matchSnapshot: {}
