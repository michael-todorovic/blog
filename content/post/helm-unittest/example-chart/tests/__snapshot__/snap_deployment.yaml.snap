test base:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-chart
        app.kubernetes.io/version: 1.16.0
        helm.sh/chart: example-chart-0.1.0
      name: RELEASE-NAME-example-chart
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: example-chart
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: example-chart
            app.kubernetes.io/version: 1.16.0
            helm.sh/chart: example-chart-0.1.0
        spec:
          containers:
            - image: nginx:1.16.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: http
              name: example-chart
              ports:
                - containerPort: 8080
                  name: http
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: http
          serviceAccountName: RELEASE-NAME-example-chart
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-chart
        app.kubernetes.io/version: 1.16.0
        helm.sh/chart: example-chart-0.1.0
      name: RELEASE-NAME-example-chart
    spec:
      ports:
        - name: http
          port: 8080
          protocol: TCP
          targetPort: http
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: example-chart
      type: ClusterIP
